stdlib/universe/universe.flux

```diff
@@ -4643,3 +4643,126 @@ toTime = (tables=<-) => tables |> map(fn: (r) => ({r with _value: time(v: r._val
 // tags: date/time
 //
 today = () => date.truncate(t: now(), unit: 1d)
+
+// yesterday finds the starting and stopping point of yesterday truncated to the day.
+//
+// ## Examples
+//
+// ### Returns a recording containing the timestamp of the start and the stopping point of yesterday
+// ```no_run
+// option now = () => 2022-01-02T13:45:28Z
+//
+// yesterday()
+// // Returns {start: 2022-01-01T00:00:00.000000000Z, stop: 2022-01-02T00:00:00.000000000Z}
+// ```
+//
+// ### Query data from yesterday
+// ```no_run
+// from(bucket: "example-bucket")
+//     |> range(start: yesterday())
+// ```
+//
+// ## Metadata
+// introduced: NEXT
+// tags: date/time
+
+yesterday = () => {
+    ret = {start: (date.sub(d:1d, from:today())), stop: today()}
+    return ret
+}
+
+
+// Monday will return the a record containing the start and the stopping point of the last monday
+// ## Parameters
+// - d: time.
+//
+// ## Examples
+//
+// ### Find the Monday of that week 
+//
+// ```no_run
+// import "date"
+//
+// d = date.monday(d: 2022-05-10T10:10:00Z)
+//
+// // Returns {start: 2022-05-08T00:00:00.000000000Z stop:2022-05-09T00:00:00.000000000Z }
+// ```
+//
+// ## Metadata
+// tags: date/time
+// 
+monday = (d) => {
+    today_date = date.truncate(t: d, unit: 1d )
+    cur_day = date.weekDay(t: today_date)
+    scaled_offset = if cur_day == date.Monday then date.scale(d:1w, n: 1)  else date.scale(d: 1d, n: (cur_day - 1))
+    mon =  date.sub(d: scaled_offset, from: today_date)
+    ret = {start: mon, stop: date.add(d: 1d, to: mon)}
+    return ret
+}
+
+
+
+// monthStart will return the timestamps of the start and end of the month in a record
+// ## Parameters
+// - d: Timestamp to find the months start for.
+//
+// ## Examples
+//
+// ### Find the timestamps for the start and stopping point of that month
+//
+// ```no_run
+// import "date"
+//
+// d = date.monthStart(d: 2022-05-10T10:10:00Z)
+//
+// // Returns {start:2022-05-01T00:00:00.000000000Z, stop:2022-06-01T00:00:00.000000000Z}
+// ```
+//
+// ## Metadata
+// tags: date/time
+// 
+monthStart = (d) => {
```

Comment:
We'll probably want to rename these `week()` and `month()` now that they are talking about the block of time they represent.